### DISCUSSION TITLE
可否在test目录下加一个演示使用c++ pkg的示例，主要是怎么build

### DISCUSSION THREAD

**Discussion by constly** [ejoy/ant#82] [Q&A] [answered]
如题，虽然年后公开的项目里面肯定会有c++的pkg，但觉得太久了.. 

我不太明白怎么做才算是比较合理的工作流，我是：
1. 在我自己的项目里面建立make.lua
2. 使用引擎的make.lua，然后我把我项目的路径加里面去（感觉这样不利于维护）

如果是第1种，我的make应该怎么写呢

**Comment by actboy168** [comment, marked as answer]
``` lua
local lm = require "luamake"
lm.AntDir = lm:path "3rd/ant"
lm:import(lm.AntDir .. "/make.lua")
lm:default {
    "bgfx-lib",
}
```

**Reply by constly** [reply]
大佬好，

1. 我按上面的要求建立了工程目录
2. 执行: luamake -mode debug runtime
3. 生成了 bin/msvc/debug/ant.exe，我双击后，提示找不到fmod.dll,我又去引擎那边拷贝了一份过来
4. 再次点击，报：无法在C盘用户目录下找到 ant/bundle/00.zip
5. 我改为执行: luamake -mode debug editor， 最后生成了lua.exe
6. 执行："bin/msvc/debug/lua.exe" "main.lua"
7. 报：cannot open engine/bootstrap.lua: No such file or directory

请问下，这种情况下，应该怎么启动项目呢~

**Reply by constly** [reply]
我把main.lua的前几行改为这样
```
package.path = "3rd/ant/engine/?.lua"
require "bootstrap"
import_package "ant.window".start {
    feature = {
       “game",
       "ant.render",
        "ant.animation",
   }
}
```
会报这个错： Preinit error: (vfs):15: cannot open engine/debugger.lua: No such file or directory

**Reply by actboy168** [reply]
启动的代码不要随便改，照抄就行

**Reply by constly** [reply]
我改的是这句：package.path = "3rd/ant/engine/?.lua"
如果不改，会报这个错： cannot open engine/bootstrap.lua: No such file or directory

**Reply by actboy168** [reply]
你需要把cwd设置为引擎目录，例如
``` bash
cd 3rd/ant && ./lua main.lua
```

**Comment by constly** [comment]
> 你需要把cwd设置为引擎目录，例如
> 
> ```shell
> cd 3rd/ant && ./lua main.lua
> ```

我这样改了我的启动指令：
```
chcp 65001
cd 3rd/ant
"../../bin/msvc/debug/lua.exe" "../../main.lua"
```
有两个问题:
1. 我发现我必须把 main.lua 放在 bin/msvc/debug/目录下， 即跟lua.exe同级，不然就找不到文件
2. 我必须把engine/game/init.lua里面11行的arg[0] 改为 arg[1]，
   不然会报: engine/game/init.lua:11: bad argument # 1 to 'path' (bee::path expected, got nil)

不知道是不是因为我的启动指令不对导致的。总之我把上面两个问题改了后确实跑起来了！接下来我试试写一个c++的pkg
